addons/hr_timesheet/migrations/10.0.1.0/pre-migration.py

```diff
@@ -11,6 +11,25 @@
 }
 
 
+def create_and_populate_department(cr):
+    cr.execute('''
+       ALTER TABLE account_analytic_line ADD COLUMN department_id INT;
+       ALTER TABLE account_analytic_line
+          ADD CONSTRAINT account_analytic_line_department_id_fkey
+          FOREIGN KEY (department_id) REFERENCES hr_department (id);
+
+       WITH departments AS (
```

Comment:
Is this subquery needed?

Reply:
I think this CTE is a very elegant solution. A subquery would be bad indeed

Reply:
Since a single user_id can be related with more than one resource, it can also have many employees, and thus belong to many departments.  This is why I even used a MAX, because I don't care for the department.  

I would have to look how does Odoo decides which department to assign using the related when it finds more that one employee.

This is probably an error, but the model allows for it.  In fact, a test in Odoo does set two employees to the same user.